main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala

```diff
@@ -1908,9 +1908,10 @@ object Weeder2 {
       flatMapN(pickNameIdent(tree), exprs) {
         case (ident, channel :: body :: Nil) =>
           Validation.success(SelectChannelRule(ident, channel, body))
-        case _ =>
+        case _ => // Unreachable case, since this is a parser error
```

Comment:
Are you sure the case is unreachable? I don't think it is... 

I would guess if you write a partial select rule it is reachable.

If it was truely unreachable then we should just throw ICE

Reply:
I added tests for that but they all failed with parser failures